src/spikeinterface/extractors/cellexplorersortingextractor.py

```diff
@@ -1,136 +1,163 @@
+from __future__ import annotations
+
 import numpy as np
 from pathlib import Path
-from typing import Union, Optional
 
 from ..core import BaseSorting, BaseSortingSegment
 from ..core.core_tools import define_function_from_class
 
 
-try:
-    import scipy.io 
-    import hdf5storage
-    HAVE_SCIPY_AND_HDF5STORAGE = True
-except ImportError:
-    HAVE_SCIPY_AND_HDF5STORAGE = False
-
-
-PathType = Union[str, Path]
-OptionalPathType = Optional[PathType]  
-
-
 class CellExplorerSortingExtractor(BaseSorting):
     """
     Extracts spiking information from .mat files stored in the CellExplorer format.
-    Spike times are stored in units of seconds.
+    Spike times are stored in units of seconds so we transform them to units of samples.
+
+    The newer version of the format is described here:
+    https://cellexplorer.org/data-structure/
+
+    Whereas the old format is described here:
+    https://github.com/buzsakilab/buzcode/wiki/Data-Formatting-Standards
 
     Parameters
     ----------
-    spikes_matfile_path : PathType
-        Path to the sorting_id.spikes.cellinfo.mat file.
+    folder_path: str | Path
+        Path to the folder containing the .mat files from the session.
+    sampling_frequency: float | None, optional
+        The sampling frequency of the data. If None, it will be extracted from the files.
+    session_info_matfile_path: str | Path | None, optional
+        Path to the session info .mat file. If None, it will be inferred from the folder path.
     """
 
     extractor_name = "CellExplorerSortingExtractor"
-    installed = HAVE_SCIPY_AND_HDF5STORAGE
     is_writable = True
-    mode = "file"
-    installation_mesg = "To use the CellExplorerSortingExtractor install scipy and hdf5storage: \n\n pip install scipy  hdf5storage"
+    mode = "folder"
+    installation_mesg = "To use the CellExplorerSortingExtractor install scipy and h5py"
+
+    def __init__(
+        self,
+        folder_path: str | Path,
```

Comment:
Does the `|` operator properly equate to `Union` when using annotations?

Reply:
Yep.

Reply:
https://peps.python.org/pep-0604/

Reply:
Nice!

Reply:
Note that you need `from __future__ import annotations` at the top of the file before python 3.10 though.